config/routes.rb

```diff
@@ -1,9 +1,20 @@
 Rails.application.routes.draw do
 
   root "users#index"
-  resources :users, only: %i[show index]
-  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
-  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+  resources :users, only: [:show] do
+    resources :posts, only: [:index, :show]
+  end
+
+  resources :posts, only: [:new, :create] do
+    resources :comments, only: [:new, :create]
+    resources :likes, only: [:create]
+  end
+
+
+
+  # resources :users, only: %i[show index]
+  # get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  # get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
 
```

Comment:
- [ ] Kindly avoid leaving commented pieces of code in your codebase. It makes your codebase unclean and is not a good practice :+1: